parquet-hadoop/src/main/java/org/apache/parquet/crypto/AesCipher.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.crypto;
+
+
+import javax.crypto.Cipher;
+import javax.security.auth.DestroyFailedException;
+
+import org.apache.parquet.ShouldNeverHappenException;
+import org.apache.parquet.crypto.ModuleCipherFactory.ModuleType;
+
+import java.io.IOException;
+import java.security.SecureRandom;
+
+
+public class AesCipher {
+
+  public static final int NONCE_LENGTH = 12;
+  public static final int GCM_TAG_LENGTH = 16;
+
+  static final int AAD_FILE_UNIQUE_LENGTH = 8;
+
+  protected static final int CTR_IV_LENGTH = 16;
+  protected static final int GCM_TAG_LENGTH_BITS = 8 * GCM_TAG_LENGTH;
+  protected static final int CHUNK_LENGTH = 4 * 1024;
+  protected static final int SIZE_LENGTH = ModuleCipherFactory.SIZE_LENGTH;
+
+  protected EncryptionKey aesKey;
+  protected final SecureRandom randomGenerator;
+  protected Cipher cipher;
+  protected final byte[] localNonce;
+  protected boolean wipedOut;
+
+
+  AesCipher(AesMode mode, byte[] keyBytes) throws IllegalArgumentException, IOException {
+    if (null == keyBytes) {
```

Comment:
Is empty keyBytes allowed?

Reply:
you probably mean all-zero keyBytes. While this is allowed, it might be problematic in certain rare situations. To be on the safe side, I've added throwing an exception for such keys.